se_shared/str_utils.h

```diff
@@ -13,12 +13,10 @@
  *
  * \param[in] s The string to split.
  * \param[in] delim The delimiter to use.
- * \param[out] elems The vector that will contain the substrings.
- * \return A reference to the vector containing the substrings.
+ * \return The vector containing the substrings.
  */
-static std::vector<std::string>& splitString(const std::string&        s,
-                                             char                      delim,
-                                             std::vector<std::string>& elems) {
+static std::vector<std::string> splitString(const std::string& s, char delim) {
+  std::vector<std::string> elems;
   std::stringstream ss(s);
```

Comment:
this also const?

Reply:
`ss` can't be made const because `std::getline` may modify its internal state (see [here](http://www.cplusplus.com/reference/string/string/getline/)).

Reply:
I realised later. Sorry about this :)